config/config.go

```diff
@@ -192,11 +165,87 @@ func loadForFileFromBytes(confBytes []byte, filePath string, kmsEncryptionContex
 		}
 		groups = append(groups, keyGroup)
 	}
+	return groups, nil
+}
+
+func loadForFileFromBytes(confBytes []byte, filePath string, kmsEncryptionContext map[string]*string, getFromDestinationRule bool) (*Config, error) {
+	conf := configFile{}
+	err := conf.load(confBytes)
+	if err != nil {
+		return nil, fmt.Errorf("error loading config: %s", err)
+	}
+
+	var rule *creationRule
+	var dRule *destinationRule
+
+	if getFromDestinationRule {
```

Comment:
I would make this two separate functions instead of one with this branching. You can extract common code into helper functions if needed